### PR TITLE
group output by test project

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: run-tests.ps1 [hunk 1]
```diff
@@ -58,26 +74,36 @@ foreach ($framework in $testFrameworks) {
...
+        if (-not $project.Value.Contains($framework)) {
+            Write-Output "Skipping $(Split-Path $project.Name -leaf) ($framework on $platform)"
+            continue
+        }
 
-    Write-Output "::endgroup::"
+        Write-Output "::group::Running $(Split-Path $project.Name -leaf) ($framework on $platform)"
+
+        # -m:1 parameter prevents test projects from being run in parallel, which could cause conflicts since PessimisticLocks project shares same tests
+        dotnet test $project.Name --configuration Release --no-build --framework $framework --logger "GitHubActions;report-warnings=false" -m:1
```
Comment (DavidBoike) [inline run-tests.ps1]: The "PessimisticLocks" comment was a specific thing in Raven, so kind of funny I forgot it here. But if we're doing these one at a time anyway, isn't `-m:1` then pointless?

```suggestion
        dotnet test $project.Name --configuration Release --no-build --framework $framework --logger "GitHubActions;report-warnings=false"
```

**Comment by DavidBoike** [comment]
I did a [spike in SQS Transport](https://github.com/Particular/NServiceBus.AmazonSQS/pull/1238) since it has a command line tests project that has different target frameworks than everything else. Turned out great ... well, aside from the test failure but that's beside the point.

image

**Review by DavidBoike — APPROVED** [review]

**Comment by bording** [comment]
From that screenshot, I see that we're now showing per-project frameworks, so what's the point of the `Detected target frameworks` section? Seems redundant.

**Comment by adamralph** [comment]
@bording yeah, I wondered about that too. Initially I thought I'd leave it in in case anyone found it useful. I also think that the "Detected test projects" section can go too. For me neither of them add any value.

Any objections to removing both?

**Comment by DavidBoike** [comment]
Fine with me, was mostly there for me to check myself when I was originally creating the action.

**Comment by adamralph** [comment]
I kicked off the TMT build again with the summary sections gone:

image

**Review by DavidBoike — APPROVED** [review]

**Comment by bording** [comment]
Seems like the "skipping" lines are kind of noise as well? Why do we need to be told that we aren't running tests for a TFM that isn't in the project?

And why do we need to be told what OS it is, when the entire actions run is going to be a single OS?

**Comment by adamralph** [comment]
@bording I've removed that noise too now. Note that since the last run, most of the projects now also target `net6.0`, so there would be very few skips anyway:

image

**Review by DavidBoike — APPROVED** [review]

**Review by bording — APPROVED** [review]

**Comment by adamralph** [comment]
FYI I've put the `v1.2.0` tag on the resulting `main` commit

**Pull request merged**